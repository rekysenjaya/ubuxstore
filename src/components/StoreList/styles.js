import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    item: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingLeft: 16,
        paddingRight: 12,
        paddingVertical: 8
    },
    store: {
        flex: 1,
        textAlign: 'left',
        paddingLeft: 8,
        fontSize: 15
    },
});

export default styles;
