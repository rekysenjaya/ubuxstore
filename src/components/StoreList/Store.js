import React, { Component } from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { Card, CardItem } from 'native-base'
import _ from 'lodash';
import styles from './styles';
import PropTypes from 'prop-types';

export default class Store extends Component {
    shouldComponentUpdate = (nextProps) => {
        if (!_.isEqual(this.props.item, nextProps.item)) {
            return true
        }
        return false
    }
    render() {
        const { tradingName, status } = this.props.item;
        return (<TouchableOpacity style={styles.gridItem} onPress={() => this.props.action(this.props.item)}>
            <Card style={styles.itemText}>
                <CardItem>
                    <Text style={styles.title} >{tradingName}</Text>
                </CardItem>
                <CardItem style={styles.statusList} >
                    <Text style={styles.status} >status: {status}</Text>
                </CardItem>
            </Card>
        </TouchableOpacity>);
    }
}

Store.propTypes = {
    item: PropTypes.object,
    action: PropTypes.func
}
