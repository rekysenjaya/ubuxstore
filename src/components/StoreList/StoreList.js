import React, { Component } from 'react';
import { FlatList } from 'react-native';
import Store from './Store';

export default class StoreList extends Component {
    renderItem = ({ item }) => {
        return <Store item={item} />;
    };

    render() {
        return (
            <FlatList
                data={this.props.data}
                renderItem={this.renderItem}
                keyExtractor={(item, index) => index}
            />
        );
    }
}
