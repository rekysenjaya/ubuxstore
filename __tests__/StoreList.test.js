/**
 * Sample React Native Snapshot Test
 */

'use strict';

import 'react-native';
import React from 'react';
import StoreList from '../src/components/StoreList';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.setTimeout(15000);

it('renders correctly', () => {
    const tree = renderer.create(<StoreList data={[]} action={() => console.log('run')} />).toJSON();
    expect(tree).toMatchSnapshot();
});