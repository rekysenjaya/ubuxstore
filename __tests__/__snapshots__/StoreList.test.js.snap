// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={Array []}
  directionalLockEnabled={true}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  getItemLayout={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  legacyImplementation={false}
  maxToRenderPerBatch={10}
  numColumns={2}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  shouldItemUpdate={[Function]}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfig={
    Object {
      "minimumViewTime": 3000,
      "viewAreaCoveragePercentThreshold": 100,
      "waitForInteraction": true,
    }
  }
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
